toothpick-runtime/src/main/java/toothpick/Toothpick.java

```diff
@@ -41,12 +41,14 @@ public static Scope openScopes(Object... names) {
     ScopeNode lastScope = null;
     for (Object name : names) {
       previousScope = lastScope;
-      lastScope = (ScopeNode) openScope(name);
+      lastScope = (ScopeNode) openScope(name, false);
       if (previousScope != null) {
         //if there was already such a node, we add a new child
         //but there might already be such a child, in that case
         //we use it.
         lastScope = previousScope.addChild(lastScope);
+      } else {
+        ConfigurationHolder.configuration.checkMultipleRootsInTpScopeForest(lastScope);
```

Comment:
What if instead of checking `if (previousScope != null) ` all the time, you create a normal `for loop` and handle the initial element at the beginning:

```java
public static Scope openScopes(Object... names) {
     if (names == null || names.length == 0) {
       throw new IllegalArgumentException("null scopes can't be open.");
     }
 
     ScopeNode lastScope = null;
     ScopeNode previousScope = (ScopeNode) openScope(names[0], true);
     for (int i = 1; i < names.length; i++) {
        lastScope = (ScopeNode) openScope(names[i], false);
        lastScope = previousScope.addChild(lastScope);
        previousScope = lastScope;
     }
  
     return lastScope;
   }
```